game-headed/src/main/java/org/triplea/game/client/HeadedGameRunner.java

```diff
@@ -103,10 +106,17 @@ public static void main(final String[] args) {
   }
 
   private static Injections constructInjections() {
-    final ServiceLoader<AiProvider> loader = ServiceLoader.load(AiProvider.class);
     return Injections.builder()
         .engineVersion(new ProductVersionReader().getVersion())
-        .aiProviders(StreamSupport.stream(loader.spliterator(), false).collect(Collectors.toList()))
+        .playerTypes(gatherPlayerTypes())
         .build();
   }
+
+  private static Collection<PlayerTypes.Type> gatherPlayerTypes() {
+    return Stream.concat(
+            StreamSupport.stream(ServiceLoader.load(AiProvider.class).spliterator(), false)
+                .map(PlayerTypes.AiType::new),
+            PlayerTypes.getBuiltInPlayerTypes().stream())
```

Comment:
I was hoping we could have that explicit list of player types here.

I'm surprised to see 'flow field' is not part of the built-in, which makes me wonder what built-in really means.

Reply:
"Built-in" are players that are in "game-core".  They are handled slightly different from ai players in the "ai" project.

Reply:
Could you explain why and how they are handled slightly different? ProAI AFAIK and DoNothing should be things we can migrate into the ai subproject. The other ones are magic and would be deleted once we've fixed up game-core to not depend on AIs for core logic.

Reply:
They can at some point be migrated to the ai subproject.  But until that is done, they have to be treated differently from the flowfield ai because of how the `PlayerTypes.Type` is built for them.

Reply:
Okay, I'll see if I can work on this at some point to improve it further, until now I think we can both see the objective but it's not necessarily feasible to get there right away.